Question2.java

```diff
@@ -0,0 +1,137 @@
+//SinglyLinkedList class
+class SinglyLinkedList<Type>
+{
+    Node head;
+
+    private class Node //Create node
+    {
+        Type data;
+        Node next;
+    }
+
+    public void add_end(Type data) //Add to end of list
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = null;
+
+        if(head==null) head = node;
+
+        else
+        {
+            Node cursor = head;
+            while(cursor.next!=null)
+            {
+                cursor = cursor.next;
+            }
+            cursor.next = node;
+        }
+
+    }
+    public void add_start(Type data) //Add to start of list
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = head;
+        head = node;
+    }
+
+    public void add_after(int index,Type data) //Add after an element
+    {
+        Node node = new Node();
+        node.data = data;
+        node.next = null;
+
+        Node cursor = head;
+        for(int i=0;i<=index-1;i++)
+        {
+            cursor = cursor.next;
+        }
+        node.next = cursor.next;
+        cursor.next = node;
+    }
+
+    public void delete_start() //Delete from start of list
+    {
+        head = head.next;
+    }
+
+    public void delete_end() //Delete from end of list
```

Comment:
This will need 2 iterations for deleting the node. Can we do it in a single iteration?